plugin-animation/src/main/java/com/mapbox/maps/plugin/animation/CameraAnimationsPluginImpl.kt

```diff
@@ -181,17 +206,55 @@ class CameraAnimationsPluginImpl : CameraAnimationsPlugin {
     handler.removeCallbacks(commitChangesRunnable)
   }
 
+  /**
+   * Assuming currentCameraState always reflects valid current camera core state -
+   * we may skip calling native setCamera in situations when given cameraOptions are part
+   * of already applied camera.
+   */
+  private fun skipNativeSetCamera(cameraOptions: CameraOptions): Boolean {
+    if (cameraOptions.isEmpty) {
+      return true
+    }
+    if (cameraOptions.anchor != null) {
+      return false
+    }
+    cameraOptions.pitch?.let { userPitch ->
+      // use-case when pitch >= 60 and terrain enabled might result in some optimizations
+      // in gl-native and different result camera state - we check just for the pitch as
+      // checking for terrain enabled requires having the style object and more complex code
+      if (userPitch >= 60.0 || userPitch != pitch) return false
+    }
+    if (cameraOptions.zoom != null && cameraOptions.zoom != zoom) {
+      return false
+    }
+    if (cameraOptions.bearing != null && cameraOptions.bearing != bearing) {
+      return false
+    }
+    if (cameraOptions.center != null && cameraOptions.center != center) {
+      return false
+    }
+    if (cameraOptions.padding != null && cameraOptions.padding != padding) {
+      return false
+    }
+    return true
+  }
```

Comment:
please add unit tests to cover these logic

Reply:
Will do

Reply:
Added